internal/sigstore/rekor.go

```diff
@@ -0,0 +1,169 @@
+// a single artifact in Rekor.
+func (r *Rekor) SearchByHash(ctx context.Context, hash string) ([]string, error) {
+	params := index.NewSearchIndexParamsWithContext(ctx)
+	params.SetQuery(
+		&models.SearchIndex{
+			Hash: hash,
+		},
+	)
+
+	index, err := r.client.Index.SearchIndex(params)
+	if err != nil {
+		return nil, fmt.Errorf("unable to search index: %w", err)
+	}
+	if !index.IsSuccess() {
+		return nil, fmt.Errorf("search failed: %s", index.Error())
+	}
+
+	return index.GetPayload(), nil
+}
+
+// GetEntry downloads entries for the provided UUID.
+func (r *Rekor) GetEntry(ctx context.Context, uuid string) (models.LogEntryAnon, error) {
+	params := entries.NewGetLogEntryByUUIDParamsWithContext(ctx)
+	params.SetEntryUUID(uuid)
+
+	entry, err := r.client.Entries.GetLogEntryByUUID(params)
+	if err != nil {
+		return models.LogEntryAnon{}, fmt.Errorf("error getting entry: %w", err)
+	}
+	if !entry.IsSuccess() {
+		return models.LogEntryAnon{}, fmt.Errorf("entries failure: %s", entry.Error())
+	}
+
+	if entires := len(entry.GetPayload()); entires != 1 {
+		return models.LogEntryAnon{}, fmt.Errorf("excepted 1 entry, but rekor returned %d", entires)
+	}
+
+	for key := range entry.Payload {
+		return entry.Payload[key], nil
+	}
+
+	return models.LogEntryAnon{}, fmt.Errorf("no entry returned")
+}
+
+// VerifyEntry performs inclusion proof verification, SignedEntryTimestamp
+// verification, and checkpoint verification of the provided entry in Rekor.
+func (r *Rekor) VerifyEntry(ctx context.Context, entry models.LogEntryAnon) (bool, error) {
+	keyResp, err := r.client.Pubkey.GetPublicKey(nil)
+	if err != nil {
+		return false, err
+	}
+	publicKey := keyResp.Payload
+
+	block, _ := pem.Decode([]byte(publicKey))
+	if block == nil {
+		return false, errors.New("failed to decode key")
+	}
+	pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return false, err
+	}
+
+	verifier, err := signature.LoadVerifier(pub, crypto.SHA256)
+	if err != nil {
+		return false, err
+	}
+
+	err = verify.VerifyLogEntry(ctx, &entry, verifier)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// GetAndVerifyEntry performs both GetEntry and VerifyEntry.
+func (r *Rekor) GetAndVerifyEntry(ctx context.Context, uuid string) (models.LogEntryAnon, bool, error) {
+	entry, err := r.GetEntry(ctx, uuid)
+	if err != nil {
+		return models.LogEntryAnon{}, false, err
+	}
+
+	valid, err := r.VerifyEntry(ctx, entry)
+	if err != nil {
+		return models.LogEntryAnon{}, valid, err
+	}
+
+	return entry, valid, nil
+}
+
+// HashedRekordFromEntry extract the base64 encoded polymorphic Body field
+// and unmarshals the contained JSON into the correct type.
+func HashedRekordFromEntry(entry models.LogEntryAnon) (*hashedrekord.V001Entry, error) {
+	var rekord models.Hashedrekord
+
+	decoded, err := base64.StdEncoding.DecodeString(entry.Body.(string))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	body, ok := entry.Body.(string)
	if !ok {
		return nil, errors.New("body is not a string")
	}
	decoded, err := base64.StdEncoding.DecodeString(body)
	if err != nil {
		return nil, err
	}
```
Adding a check if we can successfully cast the interface here.